app/src/main/java/woowacourse/shopping/data/goods/repository/GoodsRemoteDataSourceImpl.kt

```
@@ -0,0 +1,180 @@
+                        val jsonString = responseBody.string()
+                        if (jsonString.isNotEmpty()) {
+                            try {
+                                val jsonObject = gson.fromJson(jsonString, JsonObject::class.java)
+                                val productsJsonArray = jsonObject.getAsJsonArray(PRODUCTS)
+                                val listType = object : TypeToken<List<GoodsDto>>() {}.type
+                                val goodsDtoList: List<GoodsDto> =
+                                    gson.fromJson(productsJsonArray, listType)
+                                val goods = goodsDtoList.toDomain()
+
+                                onComplete(goods)
+                            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)